### PR TITLE
build: refactor circleci so sonar shows latest test results again AND remove formatcheck precommit hook

### PR DESCRIPTION
**Pull request by mswertz**
main change: reenable sonar upload. Further refactored circle ci to reduce overhead by merging prepare and build step.

side effect: removed pre-commit format check to also simplify the apps/build.gradle code. You should either always format on pre-commit or don't use it at all (and manually run the 'format' command). Updated manual.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -56,11 +56,7 @@ jobs:
   build:
     docker:
     - image: molgenis/ci-build:latest
-    - image: postgres:15-alpine
```
Comment (mswertz) [inline .circleci/config.yml:59]: not needed for build

File: .circleci/config.yml [hunk 2]
```diff
@@ -75,8 +71,12 @@ jobs:
     - attach_workspace:
         at: ~/repo
 
-    - run: ./gradlew clean assemble testClasses ci
-      
+    - run:
+        name: build jar and compile test classes and export 'ci' config
+        command: |
+          ./gradlew clean testClasses shadowJar ci
```
Comment (mswertz) [inline .circleci/config.yml:77]: shadowJar is the 'assemble' we need

**Comment by mswertz** [comment]
I am not entirely satisfied: javascriptBuild is now run twice, both during 'build' and during 'test'.

**Review by connoratrug — APPROVED** [review]
Looks good to me, ⚠️  check master flow on merge

**Pull request merged**